### PR TITLE
fix EventCpu deadlock

### PR DESCRIPTION
**Pull request by BenjaminW3**
#388
@psychocoderHPC Please give this a try ;-)

### CODE REVIEW

**Comment by psychocoderHPC** [comment]
you are fast! willtry it tomorrow thx

File: include/alpaka/event/EventCpu.hpp [hunk 1]
```diff
@@ -509,16 +530,25 @@ namespace alpaka
                     auto spEventCpuImpl(event.m_spEventCpuImpl);
 
                     std::lock_guard<std::mutex> lk(spEventCpuImpl->m_Mutex);
+
                     if(!spEventCpuImpl->m_bIsReady)
                     {
                         spEventCpuImpl->m_bIsWaitedFor = true;
+
+                        auto const enqueueCount = spEventCpuImpl->m_enqueueCount;
```
Comment (psychocoderHPC) [inline include/alpaka/event/EventCpu.hpp]: CI is crashing with errror `enqueueCount is unsed`. 

The error is wrong and we can workaround it by using `boost::ignore_unsed`

**Comment by psychocoderHPC** [comment]
I tested the fix by coping the full class to the current alpaka version (0.2.0) used within PIConGPU.
**The issue is solved with this fix**. As soon as this can be merged I will update PIConGPU.

File: include/alpaka/event/EventCpu.hpp (hunk 1 above)
Comment (BenjaminW3) [reply include/alpaka/event/EventCpu.hpp]: solved by moving line into the #ifdef

**Comment by BenjaminW3** [comment]
Ready to be approved / merged!

**Pull request merged**